### PR TITLE
Handle local errors with the ignore list

### PR DESCRIPTION
**Pull request by TheOneRing**
Fixes: #9208, #9133

Might also fix https://github.com/owncloud/client/issues/9076

### CODE REVIEW

File: src/gui/folder.cpp [hunk 1]
```diff
@@ -95,6 +95,9 @@ Folder::Folder(const FolderDefinition &definition,
 
     _syncResult.setFolder(_definition.alias);
 
+    // those errors should not persist over sessions
+    _journal.wipeErrorBlacklistCategory(SyncJournalErrorBlacklistRecord::Category::LocalSoftError);
```
Comment (dragotin) [inline src/gui/folder.cpp:99]: Would somehow be cool if we had a logging if there were blacklisted errors from the previous run. That might be an important hint when debugging. No idea how difficult that is...

File: src/libsync/owncloudpropagator.cpp [hunk 2]
```diff
@@ -213,24 +209,13 @@ static void blacklistUpdate(SyncJournalDb *journal, SyncFileItem &item)
...
     }
 
-    // Some soft errors might become louder on repeat occurrence
-    if (item._status == SyncFileItem::SoftError
-        && newEntry._retryCount > 1) {
-        qCWarning(lcPropagator) << "escalating soft error on " << item._file
-                                << " to normal error, " << item._httpErrorCode;
-        item._status = SyncFileItem::NormalError;
-        return;
-    }
```
Comment (dragotin) [inline src/libsync/owncloudpropagator.cpp:233]: Hmm, this is going away completely? Which soft errors were ending up here, that were escalated to NormalErrors before?

**Review by dragotin — APPROVED** [review]
LGTM, only the second comment pls.

File: src/gui/folder.cpp (hunk 1 above)
Comment (TheOneRing) [reply src/gui/folder.cpp:99]: Hm true, in general `wipeErrorBlacklist*` should log what was wiped

File: src/libsync/owncloudpropagator.cpp (hunk 2 above)
Comment (TheOneRing) [reply src/libsync/owncloudpropagator.cpp:233]: Hmm maybe I should look for _httpErrorCode that way http errors will behave as before.

**Review by dragotin — APPROVED** [review]
LGTM

**Pull request merged**